R/loopy-helpers.R

```diff
@@ -175,55 +182,80 @@ has_tailor_estimated <- function(x) {
 # ------------------------------------------------------------------------------
 # Prediction and postprocessing
 
-# TODO add eval_time
-# Basic prediction on a data set (holdout of calibration) including submodels.
-# Note that if there are submodels, the results only have the grid values for
-# the submodel parameter (not the whole grid).
-sched_predict_wrapper <- function(
-  sched, # TODO rename this to be more informative with what stage
-  wflow_current,
-  static,
-  estimation = FALSE # TODO rename this to be more informative
-) {
+train_post <- function(wflow_current, predictions, grid = NULL) {
+  if (is.null(grid)) {
+    grid <- tibble()
+  }
+
+  post_obj <- hardhat::extract_postprocessor(wflow_current) |>
+    tune::finalize_tailor(grid)
+
   outputs <- get_output_columns(wflow_current, syms = TRUE)
 
-  has_submodel <- has_sub_param(sched$predict_stage[[1]])
-  if (has_submodel) {
-    sub_param <- get_sub_param(sched$predict_stage[[1]])
-    sub_list <- sched$predict_stage[[1]] |>
-      dplyr::select(dplyr::all_of(sub_param)) |>
-      as.list()
-  } else {
-    sub_list <- NULL
-    sub_param <- character(0)
-  }
+  post_obj <- post_obj |>
+    fit(
+      .data = predictions,
+      outcome = !!outputs$outcome[[1]],
+      estimate = !!outputs$estimate[[1]],
+      probabilities = c(!!!outputs$probabilities)
+    )
 
-  if (estimation & static$post_estimation) {
+  post_obj
+}
+
+# ------------------------------------------------------------------------------
+
+predict_all_types <- function(
+    wflow_fit,
+    static,
+    submodel_grid = NULL,
+    predictee = "assessment"
+) {
+  predictee <- rlang::arg_match(predictee, c("assessment", "calibration"))
+  outputs <- get_output_columns(wflow_fit, syms = TRUE)
```

Comment:
Looking at `get_output_columns()` i noticed that the `syms` argument doesn't do anything.

I'll open an issue for it